src/main/scala/edu/ie3/simona/service/primary/PrimaryServiceWorker.scala

```diff
@@ -78,15 +80,30 @@ final case class PrimaryServiceWorker[V <: Value](
         Try {
           /* Set up source and acquire information */
           val factory = new TimeBasedSimpleValueFactory(valueClass, timePattern)
-          val source = new CsvTimeSeriesSource(
-            csvSep,
-            directoryPath,
-            fileNamingStrategy,
-            timeSeriesUuid,
-            filePath,
-            valueClass,
-            factory
-          )
+          val source = Try(
+            new CsvTimeSeriesSource(
+              csvSep,
+              directoryPath,
+              fileNamingStrategy,
+              timeSeriesUuid,
+              filePath,
+              valueClass,
+              factory
+            )
+          ) match {
+            case Failure(exc: DateTimeParseException) =>
+              throw new IOException(
+                s"Could not initiate the time series source. Parsing ${exc.getParsedString} failed with timepattern: $timePattern. The time pattern can be configured in the configuration.",
+                exc
+              )
```

Comment:
If we want this kind of distinction, would'nt it rather make sense to have it in PSDM?
So, everything that can go wrong within *TimeSeriesSource is fully described by an exception that its constructor throws. Then, when we arrive at the level of PrimaryServiceWorker, we can then wrap the exception once more and say: "During creation of the timeseries source, an error occured." or something like that.

Reply:
Possibly solved [here](https://github.com/ie3-institute/PowerSystemDataModel/pull/999) or with an earlier PR. We should still check if `DateTimeParseException` is also wrapped in a `SourceException` within PSDM